From 53868f5ebdc569e9674c345794c6161b604ea6e3 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 27 Mar 2024 14:10:25 +0100
Subject: [PATCH] dt-bindings: dma: stm32-dma3: introduce lli-bus-interface
 property

STM32 DMA3 has up to two master ports. It always has a port 0, and could
have a port 1. Depending on hardware configuration, ports can be AHB or
AXI. Depending on where linked-list items are allocated, or depending of
the port 0 usage, the user may want to use the second port if there is one.
lli-bus-interface property can be set to 0, to select port 0, or set to 1,
to select port 1. If the property is not defined, port 0 is selected by
default.

Change-Id: Id40f7cbe72316091008c48759e359623f4dbe55d
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/370780
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/dma/stm32/st,stm32-dma3.yaml          | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

--- a/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
+++ b/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
@@ -48,6 +48,17 @@ properties:
   resets:
     maxItems: 1
 
+  power-domains:
+    maxItems: 1
+
+  lli-bus-interface:
+    description: |
+      To select port 1 for fetching linked-list items (LLIs).
+      If not defined, port 0 is used by default.
+    $ref: /schemas/types.yaml#/definitions/uint32
+    enum: [0, 1]
+    default: 0
+
   memory-region:
     description: |
       Phandle to a /reserved-memory child node assigned to STM32 DMA3 for linked-list items
@@ -57,9 +68,6 @@ properties:
       the STM32 DMA3 performance.
     maxItems: 1
 
-  power-domains:
-    maxItems: 1
-
   st,axi-max-burst-len:
     description: |
       Restrict AXI burst length in unit of beat by value specified in this property.
